docs/development.md

```diff
@@ -91,3 +91,103 @@ The `cbindgen` xtask generates multiple header files for four different modules:
 4. The types used by the C++ interpreter API, written to `slint_interpreter_internal.h`.
 
 Typically the input to `cbindgen` is within `ffi` sub-modules in the corresponding input crates to `cbindgen`. These `ffi` modules are gated with `#[cfg(feature = "ffi")]`.
+
+## Commit History & Code Reviews
+
+Linear history is preferred over merge commits. Long lived features can live in feature branches and those can be integrated
+with merge commits, of course.
+
+As a consequence, we typically integrate pull requests as "rebase and merge" or "squash and merge".
+
+During code review, consider adding small commits on top to make it easier for the reviewer and contributor to track feedback and
+how the feedback was incorporated - rebase can tend to make it harder. It's perfectly fine to then squash these commits when the
+review phase is complete and approval is given.
+
+Example:
+
+A PR consists of three commits:
+
+1. Add new widget
+2. Add documentation for new widget
+3. Change example to use new widget
+
+In the review phase, the reviewer suggests to make changes to the widget implementation and the documentation. Consider pushing
+these as follow-up fixes:
+
+1. Add new widget
+2. Add documentation for new widget
+3. Change example to use new widget
+4. Fix an issue in widget found during review
+5. Fix a typo in the documentation
+
+Finally, the PR is approved. As contributor, in your local branch, feel free to merge 4. into 1. and 5. into 2.:
+
+(commits are real, sha1s are just examples)
+
+As a first step, let's rebase our changes to make sure that there are no conflicts:
+
+```bash
+git rebase origin/master
+```
+
+This might run through without stoppping. If there are merge conflicts to be resolved, `git rebase` will stop
+and let you fix it. For instructions how to resolve the conflicts and continue, see [Resolving merge conflicts after a Git rebase](https://docs.github.com/en/get-started/using-git/resolving-merge-conflicts-after-a-git-rebase).
+
+When your branch is rebased, proceed to squash the fixup commits. Start an interactive rebase that starts at the base commit:
+
+```
+$ git rebase -i origin/master
+```
+
+This launches the configured editor with the above list of commits,
+in the order as they will committed:
+
+```
+pick 82916bc2 Add new widget
+pick e55bde4c Add documentation for new widget
+pick 9bc8d203 Change example to use new widget
+pick a6feda52 Fix an issue in widget found during review
+pick 032032dc Fix a typo in the documentation
+```
+
+Let's merge 4. into 1. and 5. into 2. by changing the above:
+
+```
+pick 82916bc2 Add new widget
+squash a6feda52 Fix an issue in widget found during review
```

Comment:
squash or fixup?

Reply:
I'll change it to fixup. Shortens the text a little then.